### ISSUE TITLE
@inheritdoc tag isn't working

### ISSUE THREAD

**Issue by feus4177** [jsdoc3/jsdoc#932]
I'm using out the jsdoc library for the first time and I really like it. The one problem I've had is the @inheritdoc tag doesn't appear to be working. I am using jsdoc 3.3.0-beta1. Here is an example of what I'm trying to do:

```javascript

/**
 * The constructor description
 *
 * @class
 * @classdesc Base class for clients.
 * @private
 * @arg {string} token - Authentication token, provider or consumer.
 * @return {Client} A new client object.
 */
function Client(token){};


/**
 * A client method
 *
 * @method
 * @arg {string} eventName - Name of the event to listen for.
 * @arg {eventCallback} fn - Callback to run when the event is heard.
 * @return {Client} The original client object making it chainable.
 */
Client.prototype.on = function on(eventName, fn){};


/**
 * The constructor description
 *
 * @class
 * @classdesc Base class for consumers.
 * @private
 * @arg {string} token - Authentication token, provider or consumer.
 * @return {Client} A new client object.
 */
function Consumer(token){};

Consumer.prototype.on = Client.prototype.on;
```

It is my understanding that the docs for the `on` method should inherit automatically, but instead `Consumer#on` is completely undocumented. If I include `/** @inheritdoc */` then `Consumer#on` is categorized as a member.

**Comment by hegemonic** [comment]
This is happening because `Consumer` isn't tagged as inheriting from `Client` in any way, so JSDoc doesn't know what to do with the `@inheritdoc` tag. (JSDoc isn't smart enough to infer the relationship from the assignment.)

If it's okay to say that `Consumer` extends `Client`, then the following works--and the docs are inherited automatically, just as you expected:

```js
/**
 * The constructor description
 *
 * @class
 * @classdesc Base class for clients.
 * @private
 * @arg {string} token - Authentication token, provider or consumer.
 * @return {Client} A new client object.
 */
function Client(token){};


/**
 * A client method
 *
 * @method
 * @arg {string} eventName - Name of the event to listen for.
 * @arg {eventCallback} fn - Callback to run when the event is heard.
 * @return {Client} The original client object making it chainable.
 */
Client.prototype.on = function on(eventName, fn){};


/**
 * The constructor description
 *
 * @class
 * @classdesc Base class for consumers.
 * @private
 * @augments {Client}
 * @arg {string} token - Authentication token, provider or consumer.
 * @return {Client} A new client object.
 */
function Consumer(token){};

Consumer.prototype.on = Client.prototype.on;
```

If you can't say that `Consumer` extends `Client` for some reason, you'll have to copy-and-paste the docs from `Client#on` to `Consumer#on`. There's another bug tracking the ability to write things like `@inheritdoc Client#on` and have them work, but right now that doesn't work.

**Comment by hegemonic** [comment]
(P.S. Glad you've been happy with JSDoc, aside from that minor snag!)

**Comment by feus4177** [comment]
Ah, my apologies, in my actual code I the @extends tag. I accidentally removed it in my slimmed down example. I had tried `@extends Client` and `@extends {Client}` and I had thought they were working because there was an extends element showing up in the resulting HTML. After having problems with the `@link` tag I discovered my syntax was wrong. What I actually needed to fix the problem was `@extends {module:myModule~Client}`. I know this syntax is pretty common but I had never worked with it before. It might help to have a section explaining how all the linking syntax works. Thanks for your help.
